frontend/src/components/pages/admin/Admin.Roles.tsx

```diff
@@ -32,22 +28,19 @@ export class AdminRoles extends Component<{}> {
         if (!api.adminInfo) return DefaultSkeleton;
         const roles = api.adminInfo.roles;
 
-        return <Table
-            size={'middle'} style={{ margin: '0', padding: '0', whiteSpace: 'nowrap' }} bordered={false}
-            showSorterTooltip={false}
-            dataSource={roles}
-            rowClassName={() => 'hoverLink'}
-            rowKey={x => x.name}
+        return <DataTable<Role>
+            enableSorting
+            data={roles}
+            size="md"
+            expandRowByClick
+            subComponent={({row: {original: role}}) => <RoleComponent role={role}/>}
             columns={[
-                { width: 1, title: 'Role Name', dataIndex: 'name', sorter: sortField('name') },
-                { title: '', render: _r => (<span></span>) },
+                {
+                    accessorKey: 'name',
+                    header: 'Role Name',
+                    size: Infinity,
```

Comment:
How does Infinity work here? does it just take all available space?

Reply:
Is thumbs up just a confirmation? :)

Reply:
Yes, it takes up all the available space, I'd need some testing with real data to see whether this works as expected in production.